### PR TITLE
Containerized proxy - copying images to internal registry

### PR DESCRIPTION
**Pull request by 0rnela**
# Description

Customers may want to deploy SUMA containerized proxy in an environment without an internet connection.
To do so, they can copy the images to an internal registry or by saving the images to a tar file.

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [x] Manager-4.3 https://github.com/uyuni-project/uyuni-docs/pull/2100
- [ ] Manager-4.2

# Links

Fixes https://github.com/SUSE/spacewalk/issues/19597 i.e. [Confluence instructions were converted to documentation](https://confluence.suse.com/pages/viewpage.action?spaceKey=~RDiasMateus&title=Copy+SUSE+Manager+Proxy+images+to+internal+registry).

### CODE REVIEW

File: modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc [hunk 1]
```diff
@@ -0,0 +1,70 @@
+[[containerized-proxy-setup-without-internet]]
+= Containerized proxy deployment without an internet connection
+
+It is possible to deploy containerized {productname} Proxy in an environment without an internet connection.
+In such case, the images are copied to an internal registry, or saved to a tar file.
+
+This procedure allows re-use of the existing minion FQDN to create a configuration to start a containerized proxy.
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc:7]: this is not related with a proxy deployment without internet connection

File: modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+Changing the base channel to have the SUSE Manager client tools can lead to package conflicts and port conflicts.
+
+
+[IMPORTANT]
+====
+This procedure applies to {salt} proxies only.
+====
+
+[[from.suma.to.internal.reg.without.ssl]]
+== Copy images from suse manager registry to internal registry without SSL configured
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc:22]:
```suggestion
== Copy images from SUSE registry to internal registry without SSL configured
```

File: modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+[NOTE]
+====
+If you want to copy the images to a registry with SSL, remove the property [literal]`--dest-tls-verify=false`.
+====
+
+. In a machine with access to "registry.suse.com" (CONFIRM: what machine - SUMA server?) install [literal]`skopeo` by using the command
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc:29]: can be any machine, no need to be on suse manager machine

File: modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+= Containerized proxy deployment without an internet connection
+
+It is possible to deploy containerized {productname} Proxy in an environment without an internet connection.
+In such case, the images are copied to an internal registry, or saved to a tar file.
+
+This procedure allows re-use of the existing minion FQDN to create a configuration to start a containerized proxy.
+The user should first register the minion to suse manager server, and then re-use the same FQDN in a generation.
+
+
+Make sure the default port is set up to `8022` for virtualized proxy when creating configuration using [literal]`spacecmd`.
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc:11]: this is not related to connected setup, but rater related to general installation.

**Review by rjmateus — COMMENTED** [review]
Some comments to be reviewed

**Comment by rjmateus** [comment]
please add also @cbosdo as a reviewer

File: modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+[[from.suma.to.internal.reg.without.ssl]]
+== Copy images from SUSE registry to internal registry without SSL configured
+
+[NOTE]
+====
+If you want to copy the images to a registry with SSL, remove the property [literal]`--dest-tls-verify=false`.
+====
+
+. In a machine with access to "registry.suse.com" (CONFIRM: what machine - SUMA server?) install [literal]`skopeo` by using the command
```
Comment (0rnela) [inline modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc:29]:
```suggestion
. In a machine with access to "registry.suse.com" install [literal]`skopeo` by using the command
```

File: modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc [hunk 6]
```diff
@@ -0,0 +1,70 @@
+[[containerized-proxy-setup-without-internet]]
+= Containerized proxy deployment without an internet connection
+
+It is possible to deploy containerized {productname} Proxy in an environment without an internet connection.
+In such case, the images are copied to an internal registry, or saved to a tar file.
+
+This procedure allows re-use of the existing minion FQDN to create a configuration to start a containerized proxy.
+The user should first register the minion to suse manager server, and then re-use the same FQDN in a generation.
```
Comment (0rnela) [inline modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc:8]:
```suggestion
The user should first register the minion to {productname} server, and then re-use the same FQDN in a generation.
```

File: modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc (hunk 3 above)
Comment (0rnela) [reply modules/installation-and-upgrade/pages/containerized-proxy-wo-internet.adoc]: @rjmateus - would this sentence work better:

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
++
+. If the registry is unsecured, i.e. not configured with SSL, add the regitry domain to the section [literal]``insecure list`` on the containerized proxy virtual machine by editing:
++
+----
+/etc/containers/registries.conf
+----
++
+[NOTE]
+====
+If you want to copy the images to a registry with SSL, remove the property [literal]`--dest-tls-verify=false`.
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:42]: Shouldn't we invert the logic and have a note to add that parameter if the registry is not secured?

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+podman save -m -o proxy-images.tar \
+    k8s.gcr.io/pause \
+    registry.suse.com/suse/manager/4.3/proxy-httpd \
+    registry.suse.com/suse/manager/4.3/proxy-salt-broker \
+    registry.suse.com/suse/manager/4.3/proxy-squid registry.suse.com/suse/manager/4.3/proxy-ssh \
+    registry.suse.com/suse/manager/4.3/proxy-tftpd
+----
++
+. Transfer the [literal]`proxy-images.tar` to the air-gapped proxy.
+. Run this command to start the pod:
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:71]: This command does not start the pod, just makes the images available for it to be started after

**Review by cbosdo — COMMENTED** [review]
We miss the tweaks needed to use these images... but there is one thing I am not sure about and need to test to provide you the proper steps.

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 9]
```diff
@@ -0,0 +1,75 @@
+[[containerized-proxy-deployment]]
+= Containerized proxy deployment
+
+It is possible to deploy containerized images in an environment without an internet connection.
+In such case, the images are copied from {suse} registry to an internal registry, or saved to a [literal]``tar`` file.
+
+You should first register Salt minion to {productname} server, and then re-use the same FQDN in a generation.
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:7]: remove this sentence, this is already said in other part of the documentation

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 10]
```diff
@@ -0,0 +1,75 @@
+[[containerized-proxy-deployment]]
+= Containerized proxy deployment
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:2]:
```suggestion
= Containerized proxy deployment using internal registry
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+
+
+
+[[image.from.suse.to.internal.registry]]
+== Image copying from {suse} registry to internal registry 
+
+
+This example illustrates deployment of {salt} proxies only.
+
+.Procedure: Deploying {salt} Proxy from containerized image
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:17]:
```suggestion
.Procedure: Deploying {salt} Proxy from an internal image registry
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
++
+----
+/etc/containers/registries.conf
+----
++
+[NOTE]
+====
+If you want to copy the images to a registry with SSL, remove the property [literal]`--dest-tls-verify=false`.
+====
+
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:44]: I think we are missing a block explaining on how to use the images from the internal registry. Something like:
```
To start using the images from the internal registry please adapt the image path at file `/etc/sysconfig/uyuni-proxy-systemd-services.config`.
```
@cbosdo looks fine?

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc (hunk 7 above)
Comment (0rnela) [reply modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:42]: @cbosdo - Would this work? "If the registry is not secured, add `--dest-tls-verify=false` ?

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 13]
```diff
@@ -0,0 +1,74 @@
+[[containerized-proxy-deployment]]
+= Containerized proxy deployment using internal registry
+
+It is possible to deploy containerized images in an environment without an internet connection.
+In such case, the images are copied from {suse} registry to an internal registry, or saved to a [literal]``tar`` file.
```
Comment (rjmateus) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:5]:
```suggestion
In such case, the images can be copied from {suse} registry to an internal registry, or saved to a [literal]``tar`` file.
```

**Review by rjmateus — APPROVED** [review]
Form my side just one small change

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc (hunk 7 above)
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:39]: yes it would

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc (hunk 12 above)
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:43]: I wonder if that is enough. I need to test it: the `pause` image may need to be changed in the config

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+----
+zypper in skopeo (can be {productname} server)
+----
++
+. Copy images between registries:
++
+----
+for image in httpd proxy-salt-broker squid ssh tftpd; do
+    skopeo copy --dest-tls-verify=false docker://registry.suse.com/suse/manager/4.3/proxy-$image:latest docker://(your_server)/proxy-$image
+done
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:28]:
```suggestion
for image in httpd salt-broker squid ssh tftpd; do
    skopeo copy docker://registry.suse.com/suse/manager/4.3/proxy-$image:latest docker://(your_server)/registry.suse.com/suse/manager/4.3/proxy-$image
done
skopeo copy docker://k8s.gcr.io/pause:latest docker://(your_server)/k8s.gcr.io/pause:latest
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc (hunk 12 above)
Comment (cbosdo) [reply modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:43]: You need to run something like this before starting the pod for `podman` to locate the `pause` image:

```
echo -e '[engine]\nimfra_image = "(your_server)/pause:latest"'>>/etc/containers/containers.conf
```

**This doesn't work on SLE 15 SP3 and before container host as it is affected by https://github.com/containers/podman/issues/12245.**

For the k3s deployment, just add `--set repository=(your_server)` to the helm install command line.

**Review by cbosdo — COMMENTED** [review]
Added more infos after testing

**Comment by 0rnela** [comment]
@cbosdo -- final check please, I've integrated your comments. Thanks in advance.

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 15]
```diff
@@ -0,0 +1,92 @@
...
++
+. If the registry is unsecured, i.e. not configured with SSL, add the regitry domain to the section [literal]``registries.insecure`` on the containerized proxy virtual machine by editing:
++
+----
+/etc/containers/registries.conf
+----
++
+[NOTE]
+====
+If the registry is not secured, add [literal]``--dest-tls-verify=false``.
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:40]: May be we want to precise this option has to be added to the `skope copy` command line above?

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 16]
```diff
@@ -0,0 +1,92 @@
...
+This example illustrates deployment of containerized image on a machine with no access to internet.
+
+
+.Procedure: Deploying air-gapped proxy
+[role=procedure]
+
+. Before starting the pod, to locate the pause image run the command:
++
+----
+echo -e '[engine]\nimfra_image = "(your_server)/pause:latest"'>>/etc/containers/containers.conf
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:57]:
```suggestion
echo -e '[engine]\ninfra_image = "(your_server)/pause:latest"'>>/etc/containers/containers.conf
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 17]
```diff
@@ -0,0 +1,92 @@
...
+[[air-gapped-solution-for-podman]]
+== Air-gapped solution for Podman
+
+This example illustrates deployment of containerized image on a machine with no access to internet.
+
+
+.Procedure: Deploying air-gapped proxy
+[role=procedure]
+
+. Before starting the pod, to locate the pause image run the command:
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:54]: Is `locate` the appropriate word here? This basically tells `podman` where to get the `pause` image from on the internal registry.

This also goes with the above section, using an internal registry.

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+----
++
+. Transfer the [literal]`proxy-images.tar` to the air-gapped proxy.
+. To make images available to be started when needed, run the command:
++
+----
+podman load -i proxy-images.tar
+----
++
+[NOTE]
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:89]: This note goes with the previous section using the internal registry.

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 19]
```diff
@@ -0,0 +1,92 @@
...
+====
+If the registry is not secured, add [literal]``--dest-tls-verify=false``.
+====
++
+. To start using the images from the internal registry please adapt the [literal]`NAMESPACE` value in file `/etc/sysconfig/uyuni-proxy-systemd-services.config`.
+
+[[air-gapped-solution-for-podman]]
+== Air-gapped solution for Podman
+
+This example illustrates deployment of containerized image on a machine with no access to internet.
```
Comment (cbosdo) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:48]: Not sure if you make a wording difference, but this procedure is for strict air-gapped setup where the Uyuni server and proxy network has no internet connection at all. This means the images need to be transferred using USB key.

Note that this doesn't prevent to have a mix of the two scenarios where the isolated network has an internal registry, but I suppose users with such setup will be smart enough to adapt the doc to their use case.

**Review by cbosdo — COMMENTED** [review]
There is some mix and match between the two solutions. I hope I have explained clearly enough.

**Comment by 0rnela** [comment]
@cbosdo - final final check please :)

**Review by cbosdo — APPROVED** [review]
Many thanks for your patience!

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 20]
```diff
@@ -0,0 +1,109 @@
+[[containerized-proxy-deployment]]
+= Containerized proxy deployment using internal registry
+
+It is possible to deploy containerized images in an environment without an internet connection.
+In such case, the images can be copied from {suse} registry to an internal registry, or saved to a [literal]``tar`` file.
+
+
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:7]:
```suggestion


```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+[[image.from.suse.to.internal.registry]]
+== Image copying from {suse} registry to internal registry 
+
+
+This example illustrates deployment of {salt} proxies only.
+
+.Procedure: Deploying {salt} Proxy from an internal image registry
+[role=procedure]
+
+. Install [literal]``skopeo`` on a machine with access to [literal]``registry.suse.com``:
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:17]: I think you'd better add "on a machine with access to [literal]``registry.suse.com``" to the sentence before the procedure.  It seems to be a requirement.
Cf. the following procedure.

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 22]
```diff
@@ -0,0 +1,109 @@
...
++
+----
+zypper in skopeo
+----
++
+[NOTE]
+====
+This can be {productname} server.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:25]:
```suggestion
This can be {productname} Server.
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 23]
```diff
@@ -0,0 +1,109 @@
...
+====
++
+. Copy images between registries:
++
+----
+for image in httpd salt-broker squid ssh tftpd; do
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:31]: I think we'd better use pointy brackets here instead of parens: Instead of `(your_server)` use `<your_server>`.

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 24]
```diff
@@ -0,0 +1,109 @@
...
+done
+skopeo copy docker://k8s.gcr.io/pause:latest docker://(your_server)/k8s.gcr.io/pause:latest
+----
++
+[NOTE]
+====
+For every [literal]`skopeo` command add [literal]`--dest-tls-verify=false` if the registry is not secured.
+====
++
+. If the registry is unsecured, i.e. not configured with SSL, add the regitry domain to the section [literal]``registries.insecure`` on the containerized proxy virtual machine by editing:
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:42]:
```suggestion
. If the registry is unsecured, for example not configured with SSL, add the registry domain to the section [literal]``registries.insecure`` on the containerized proxy virtual machine by editing:
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 25]
```diff
@@ -0,0 +1,109 @@
...
++
+----
+/etc/containers/registries.conf
+----
++
+. Before starting the pod, point the Podman where to get the [literal]`pause` image from on the internal registry:
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:48]:
```suggestion
. Before starting the pod, point the Podman where to get the [literal]``pause`` image from on the internal registry:
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 26]
```diff
@@ -0,0 +1,109 @@
...
++
+----
+echo -e '[engine]\ninfra_image = "(your_server)/pause:latest"'>>/etc/containers/containers.conf
+----
++
+. To start using the images from the internal registry please adapt the [literal]`NAMESPACE` value in file `/etc/sysconfig/uyuni-proxy-systemd-services.config`.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:54]:
```suggestion
. To start using the images from the internal registry adapt the [literal]``NAMESPACE`` value in [path]``/etc/sysconfig/uyuni-proxy-systemd-services.config``.
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 27]
```diff
@@ -0,0 +1,109 @@
...
++
+[NOTE]
+====
+For the k3s deployment, add [literal]`--set repository=(your_server)` to the helm install command line.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:58]:
```suggestion
For the k3s deployment, add [literal]``--set repository=(your_server)`` to the helm install command line.
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 28]
```diff
@@ -0,0 +1,109 @@
...
+[[air-gapped-solution-for-podman]]
+== Air-gapped solution for Podman
+
+This example illustrates deployment of containerized image on a machine with no access to internet.
+
+
+.Procedure: Deploying air-gapped proxy
+[role=procedure]
+
+. Before starting the pod, point the Podman where to get the [literal]`pause` image from on the internal registry:
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:71]:
```suggestion
. Before starting the pod, point the Podman where to get the [literal]``pause`` image from on the internal registry:
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 29]
```diff
@@ -0,0 +1,109 @@
...
++
+----
+echo -e '[engine]\ninfra_image = "(your_server)/pause:latest"'>>/etc/containers/containers.conf
+----
++
+[WARNING]
+====
+
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:79]:
```suggestion
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 30]
```diff
@@ -0,0 +1,109 @@
...
+for image in httpd salt-broker squid ssh tftpd; do
+    podman pull registry.suse.com/suse/manager/4.3/proxy-$image
+done
+podman pull k8s.gcr.io/pause
+
+podman save -m -o proxy-images.tar \
+    k8s.gcr.io/pause \
+    registry.suse.com/suse/manager/4.3/proxy-httpd \
+    registry.suse.com/suse/manager/4.3/proxy-salt-broker \
+    registry.suse.com/suse/manager/4.3/proxy-squid registry.suse.com/suse/manager/4.3/proxy-ssh \
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:95]:
```suggestion
    registry.suse.com/suse/manager/4.3/proxy-squid \
    registry.suse.com/suse/manager/4.3/proxy-ssh \
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 31]
```diff
@@ -0,0 +1,109 @@
...
+    registry.suse.com/suse/manager/4.3/proxy-tftpd
+----
++
+[NOTE]
+====
+For the k3s deployment, add [literal]`--set repository=(your_server)` to the helm install command line.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:101]:
```suggestion
For the k3s deployment, add [literal]``--set repository=(your_server)`` to the helm install command line.
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 32]
```diff
@@ -0,0 +1,109 @@
...
+====
++
+. Transfer the [literal]`proxy-images.tar` to the air-gapped proxy.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:104]:
```suggestion
. Transfer the [literal]``proxy-images.tar`` to the air-gapped proxy.
```

**Review by keichwa — CHANGES_REQUESTED** [review]
Most proposals for consistency.

**Comment by 0rnela** [comment]
> Many thanks for your patience!

Many thanks for YOUR patience too @cbosdo.

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 33]
```diff
@@ -0,0 +1,110 @@
...
+[[image.from.suse.to.internal.registry]]
+== Image copying from {suse} registry to internal registry 
+
+
+This example illustrates deployment of {salt} proxies only.
+
+.Procedure: Deploying {salt} Proxy from an internal image registry
+[role=procedure]
+
+. On a machine with access to [literal]``registry.suse.com`` install [literal]``skopeo``:
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:18]:
```suggestion
. Install [literal]``skopeo``:
```

File: modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc [hunk 34]
```diff
@@ -0,0 +1,110 @@
...
+It is possible to deploy containerized images in an environment without an internet connection.
+In such case, the images can be copied from {suse} registry to an internal registry, or saved to a [literal]``tar`` file.
+
+
+
+[[image.from.suse.to.internal.registry]]
+== Image copying from {suse} registry to internal registry 
+
+
+This example illustrates deployment of {salt} proxies only.
```
Comment (keichwa) [inline modules/installation-and-upgrade/pages/containerized-proxy-deployment.adoc:13]:
```suggestion
This example illustrates deployment of {salt} proxies only.
Machines must have access to [literal]``registry.suse.com``.
```

**Review by keichwa — APPROVED** [review]
